src/filters/multipart.rs

```diff
@@ -131,17 +123,18 @@ impl Stream for FormData {
 impl Part {
     /// Get the name of this part.
     pub fn name(&self) -> &str {
-        &self.headers.name
+        self.part.name().expect("checked for name previously")
```

Comment:
Is the check missing? I'd expect it in `poll_next`, after matching that `Some(part)` was found. If the name is missing, then return `Err()`.

Reply:
I would change it to:

```rust
Poll::Ready(Ok(Some(part))) => {
    if part.name().is_some() {
        Poll::Ready(Some(Ok(Part { part })))
    } else {
        Poll::Ready(Some(Err(crate::Error::new(MultipartFieldMissingName))))
    }
}
```

And define lower in the field:

```rust
#[derive(Debug)]
struct MultipartFieldMissingName;

impl std::error::Error for MultipartFieldMissingName {}
```

Reply:
Sorry, I just realized that all filters have their errors contained in their files, I naively looked only on error.rs, thinking they would be grouped there.
I just sent the change.